### PR TITLE
Initial implementation of the OpenTracing bridge.

### PR DESCRIPTION
**Pull request by pcwiese**
This is, where possible, a direct port of the current Java shim. There are some holes/TODOS. Notably:

- Missing the Binary injection/extraction of the span context.  This doesn't exist with the current implementation of OpenTracing yet.
- Missing explicit timestamps for calls to OpenTracing ISpan::Log.
- Missing explicit timestamps for calls to OpenTracing ISpan::Finish. Requires implementation of https://github.com/open-telemetry/oteps/blob/master/text/0002-remove-spandata.md
- Various calls to set/get baggage are not implemented. Requires OpenTelemetry DistributedContext to be finalized.

Other oddities worth nothing:
- The SpanBuilderShim has some conditional code for calling OpenTel SetCreateChild(true|false)
- The ScopeManagerShim is actively tracking scopes returned by calls to OpenTel WithSpan and associating those with the OpenTel ISpan used when calling WithSpan.

I tested this shim against an ASP.NET Core 3 Preview 8 service wired up with OpenTracing netcore and grpc collectors and exporting to Jaeger backend.

The included tests are unit tests only; no E2E tests.

### CODE REVIEW

File: src/OpenTelemetry.Abstractions/Context/Propagation/TraceContextFormat.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
                     return SpanContext.Blank;
                 }
 
-                var traceparent = traceparentCollection.First();
+                var traceparent = traceparentCollection?.First();
```
Comment (austinlparker) [inline src/OpenTelemetry.Abstractions/Context/Propagation/TraceContextFormat.cs]: I don't understand why this change is in this PR?

Comment (lmolkova) [reply src/OpenTelemetry.Abstractions/Context/Propagation/TraceContextFormat.cs]: i think it's an accidental bugfix

**Comment by lmolkova** [comment]
it seems @pcwiese is out on the leave. I'll merge it before it diverges from master

**Comment by pcwiese** [comment]
Thanks for merging. I am on paternity for the next month or so.

**Comment by austinlparker** [comment]
Thanks @lmolkova, and congratulations @pcwiese!

**Comment by lmolkova** [comment]
@pcwiese  congratulations!
